app/scripts/components/exploration/types.d.ts.ts

```diff
@@ -7,30 +9,109 @@ export enum TimeDensity {
 export enum TimelineDatasetStatus {
   IDLE = 'idle',
   LOADING = 'loading',
-  SUCCEEDED = 'succeeded',
-  ERRORED = 'errored'
+  SUCCESS = 'success',
+  ERROR = 'error'
+}
+
+export interface StacDatasetData {
+  isPeriodic: boolean;
+  timeDensity: TimeDensity;
+  domain: string[];
 }
 
 export type AnalysisTimeseriesEntry = Record<string, number | null> & {
   date: Date;
 };
 
-export interface TimelineDatasetAnalysis {
-  status: TimelineDatasetStatus;
-  data: {
-    timeseries?: AnalysisTimeseriesEntry[];
+// TimelineDatasetAnalysis type discriminants
+export interface TimelineDatasetAnalysisIdle {
+  status: TimelineDatasetStatus.IDLE;
+  data: null;
+  error: null;
+  meta: Record<string, never>;
+}
+export interface TimelineDatasetAnalysisLoading {
+  status: TimelineDatasetStatus.LOADING;
+  data: null;
+  error: null;
+  meta: {
+    loaded?: number;
+    total?: number;
   };
-  error: any;
+}
+export interface TimelineDatasetAnalysisError {
+  status: TimelineDatasetStatus.ERROR;
+  data: null;
+  error: unknown;
   meta: {
     loaded?: number;
     total?: number;
   };
 }
+export interface TimelineDatasetAnalysisSuccess {
+  status: TimelineDatasetStatus.SUCCESS;
+  data: {
+    timeseries: AnalysisTimeseriesEntry[];
+  };
+  error: null;
+  meta: {
+    loaded: number;
+    total: number;
+  };
+}
+
+export type TimelineDatasetAnalysis =
+  | TimelineDatasetAnalysisIdle
+  | TimelineDatasetAnalysisLoading
+  | TimelineDatasetAnalysisError
+  | TimelineDatasetAnalysisSuccess;
 
-export interface TimelineDataset {
-  status: TimelineDatasetStatus;
-  data: any;
-  error: any;
+// END TimelineDatasetAnalysis type discriminants
+
+export interface TimelineDatasetData extends DatasetLayer {
+  isPeriodic: boolean;
+  timeDensity: TimeDensity;
+  domain: Date[];
+}
+
+// TimelineDataset type discriminants
+export interface TimelineDatasetIdle {
+  status: TimelineDatasetStatus.IDLE;
+  data: DatasetLayer;
+  error: null;
+  settings: {
+    // user defined settings like visibility, opacity
+    isVisible?: boolean;
+    opacity?: number;
+  };
+  analysis: TimelineDatasetAnalysisIdle;
+}
+export interface TimelineDatasetLoading {
+  status: TimelineDatasetStatus.LOADING;
+  data: DatasetLayer;
+  error: null;
+  settings: {
+    // user defined settings like visibility, opacity
+    isVisible?: boolean;
+    opacity?: number;
+  };
+  analysis: TimelineDatasetAnalysisIdle;
+}
+export interface TimelineDatasetError {
+  status: TimelineDatasetStatus.ERROR;
+  data: DatasetLayer;
+  error: unknown;
+  settings: {
+    // user defined settings like visibility, opacity
+    isVisible?: boolean;
+    opacity?: number;
+  };
+  analysis: TimelineDatasetAnalysisIdle;
+}
+export interface TimelineDatasetSuccess {
+  status: TimelineDatasetStatus.SUCCESS;
+  data: TimelineDatasetData;
+  error: null;
```

Comment:
This can be simplified by:
- wrapping settings into a common Settings typr:
```
interface Settings {
  isVisible?: boolean;
  opacity?: number;
}
```
- extend a base type for all the unioned types:
```
interface TimelineDatasetBase {
  data: DatasetLayer | TimelineDatasetData;
  error: unknown | null;
  settings: Settings;
  analysis: TimelineDatasetAnalysisIdle | TimelineDatasetAnalysis;
}
```